widget/view.controller.js

```diff
@@ -0,0 +1,146 @@
+    $scope.feedConnectors = [];
+    $scope.installedConnectors = [];
+    $scope.searchQuery = '';
+    $scope.widgetBasePath = widgetBasePath;
+    $scope.widgetCSS = widgetBasePath + 'widgetAssets/css/wizard-style.css';
+
+    init();
+
+    function init() {
+      $scope.isFortiGuardConectorInsttalled = true;
+      appModulesService.load(true).then(function (modules) {
+        modules = $filter('playbookModules')(modules);
+        $scope.modules = currentPermissionsService.availablePermissions(modules, 'create');
+      });
+    }
+
+    function toggleSelectFeedsSettings(event) {
+      event.stopPropagation();
+    }
+
+    function installConnector() {
+      WizardHandler.wizard('timSolutionpackWizard').next();
+      $scope.getSelectedConnectorsCount = function () {
+        var feedConnectors = Array.isArray($scope.feedConnectors) ? $scope.feedConnectors : [];
+        var installedConnectors = Array.isArray($scope.installedConnectors) ? $scope.installedConnectors : [];
+
+        return feedConnectors.filter(function (connector) {
+          return connector.selectConnector;
+        }).length + installedConnectors.length;
+      };
+      const selectedConnectors = $scope.feedConnectors.filter(pack => pack.selectConnector === true && pack.installed === false);
+
+      function installSequentially(index) {
+        if (index >= selectedConnectors.length) {
+          $scope.allConnectorsInstalled = true;
+          return;
+        }
+        const connector = selectedConnectors[index];
+        threatIntelManagementConfigurationService.installConnector(connector)
+          .then(resp => {
+            const importJobId = resp.data.importJob;
+            threatIntelManagementConfigurationService.getInstallationProgress(importJobId, connector)
+              .then(() => {
+                connector.installed = true;
+                installSequentially(index + 1);
+              })
+              .catch(error => {
+                console.log(error);
+                installSequentially(index + 1);
+              });
+          })
+          .catch(error => {
+            console.log(error);
+            installSequentially(index + 1);
+          });
+      }
+      installSequentially(0);
+    }
+
+    function backToStartPage() {
+      WizardHandler.wizard('timSolutionpackWizard').previous();
+    }
+
+    function backToinstallConnector() {
+      $scope.feedConnectors.sort((firstItem, secondItem) => secondItem.installed - firstItem.installed);
+      $scope.allConnectorsInstalled = false;
+      WizardHandler.wizard('timSolutionpackWizard').previous();
+    }
+
+    function moveToSelectConnector() {
+      threatIntelManagementConfigurationService.getFeedConnectors('Fortinet FortiGuard Threat Intelligence').then(function (response) {
+        $scope.fortiGuardConector = response.data['hydra:member'][0];
+        if ($scope.fortiGuardConector.installed === false) {
+          $scope.isFortiGuardConectorInsttalled = false;
```

Comment:
$scope.fortiGuardConector.installed this same property can be used instead of new variable